parlai/core/params.py

```diff
@@ -159,6 +159,8 @@ def str2floats(s):
     """
     Look for single float or comma-separated floats.
     """
+    if "[" in s:
```

Comment:
is this really necessary?

Reply:
at the very least, you need to lead a comment explaining.

Reply:
The issue is, sometimes (atleast I think its not always) a string is passed in that is of the form of a list (i.e. "[1, 2, 3]"). I am not super sure why that is the case but then the error thrown was cannot convert "[1" to float because it was splitting on comma.

I would be happy to change it if there is a better way/different place to deal with this. Would be grateful for ideas/suggestions.

Reply:
Can you help me figure out WHERE the string is being passed in? I don't think that should be the case after you switched to `_value2str`

Reply:
See [here](https://circleci.com/gh/facebookresearch/ParlAI/42246?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link). This stack trace prompted me to make the change

Reply:
I see. Look back at the rest of _kwargs_to_str_args. There's another paragraph of code that needs the same str -> value2argstr change.

Reply:
I can add a comment explaining the precaution of that if statement.

Reply:
I think after you fix the other thing, it won't be necessary to have these lines any more.

Reply:
Tried that in the latest push